src/cargo/core/compiler/mod.rs

```diff
@@ -413,7 +413,22 @@ fn rustc(cx: &mut Context<'_, '_>, unit: &Unit, exec: &Arc<dyn Executor>) -> Car
                         1 => " due to previous error".to_string(),
                         count => format!(" due to {} previous errors", count),
                     };
-                    format!("could not compile `{}`{}{}", name, errors, warnings)
+                    let mut message = format!(
```

Comment:
Thanks for doing this! Is it possible to extract this part as a function and reuse it in both warnings and errors? It would be great not to repeat ourselves. That was what I referred to originally as well.

Reply:
I've commited a new version extracting the base message building to a function, but i think _CI / Tests Linux x86_64 nightly (pull_request)_ is stuck on `sudo apt update` (>1h)

Reply:
That's strange. Perhaps a network connection error occurred.